syft/serde/msgpack/serde.py

```diff
@@ -169,6 +119,9 @@ def _force_full_simplify(worker: AbstractWorker, obj: object) -> object:
     Returns:
         The simplified object.
     """
+
+    init_global_vars_msgpack()
```

Comment:
This makes us rely on order of initialization and could create really awkard and hard to find bugs, for example, let's say that when we are loading the first time syft, syft/serde/msgpack/serde.py should be the last one to be loaded (so that all the classes should be loaded to actually get themselves enroled in the interface). I enjoy this lazy init because: 1. forces everybody to use the public API, not the global variables, 2. 100% safe, we are generating those global variables when syft is fully loaded and the interface has all it's subclasses enrolled. The other way would be to mess around in `__init__.py` files, I think it may be better to leave those alone.

Reply:
I'd also prefer if it was done in an __init__.py file but if you think it's too complex, I'm ok with leaving it for now

Reply:
These are all good points! What if the global was an object with a getter that did the lazy initialization inside? Then we wouldn't have to call the `init_global_vars` function in so many places, because it would be built in to the getter.

Reply:
My problem is here:
```
from syft.serde.syft_serializable import SyftSerializable, get_protobuf_subclasses

print(len(get_protobuf_subclasses(SyftSerializable))) #12

class Example(SyftSerializable):
    @staticmethod
    def simplify(worker, obj):
        pass

    @staticmethod
    def detail(worker, obj):
        pass

    @staticmethod
    def bufferize(worker, obj):
        pass

    @staticmethod
    def unbufferize(worker, obj):
        pass

    @staticmethod
    def get_protobuf_schema():
        pass

print(len(get_protobuf_subclasses(SyftSerializable))) #13
```
If those globals were generated and a new class is added, the globals will not reflect this, but I figured a way to do this properly I think without having inits everywhere/checks/performance impact. (currently, i'm solving this only if the classes are a part of syft, but in time that might not be the case).